server/src/send-update/send-update.controller.ts

```diff
@@ -0,0 +1,79 @@
+import { Controller, Param, Post, Req, Res } from "@nestjs/common";
+import { ConfigService } from "../config/config.service";
+import { SendUpdateService } from "./send-update.service";
+import { ProjectService } from "../project/project.service";
+import { ListservAuthService } from "../listserv/auth.service";
+import { Request } from "express";
+
+@Controller()
+export class SendUpdateController {
+  apiKey = "";
+  list = "";
+  sendgridEnvironment = "";
+
+  constructor(
+    private readonly config: ConfigService,
+    private readonly sendUpdateService: SendUpdateService,
+    private readonly projectService: ProjectService,
+    private readonly listservAuthService: ListservAuthService
+  ) {
+    this.apiKey = this.config.get("SENDGRID_API_KEY");
+    this.list = this.config.get("SENDGRID_LIST");
+    this.sendgridEnvironment = this.config.get("SENDGRID_ENVIRONMENT")
+  }
+
+  @Post("/projects/:id/send-update")
+  async sendUpdate(@Param() params, @Req() request: Request, @Res() response) {
+    const validatedUser = await this.listservAuthService.validateUser(request.headers.authorization)
+    if (!validatedUser) {
+      response.status(401).send();
+      return;
+    }
+
+    const project = await this.projectService.findOneByName(params.id);
+    // If no project is found, projectService returns HTTP error automatically, and this function does not continue
+
+    // Production names use underscores, staging use dashes
+    const segmentName = `zap${this.sendgridEnvironment === "production" ? "_production_" : "-staging-"}${project["data"]["attributes"]["dcp-borough"] === "Citywide" ? "CW" : project["data"]["attributes"]["dcp-validatedcommunitydistricts"]}`
+    const segmentIdResult = await this.sendUpdateService.getSegmentId(segmentName)
```

Comment:
A project can be "citywide" and belong to multiple CDs so this function would have to identify the _list_ of segment IDs that the project corresponds to and create a single send for each.

Reply:
My mistake, I have a better sense of the location options now.  It seems that if a user is in multiple segments receiving a single send, they will only receive one copy of the email, so it makes more sense to create one single send that goes to all of the segments, than an individual send for each segment.

Also, I want to bring to your attention, it's possible for a project to be neither Citywide nor associated with any CD, in which case the project would not fall under any of our segments. Ex. https://zap.planning.nyc.gov/projects/2024Q0218

Reply:
If a project belongs to several CDs, and a user is subbed to all of those CDs, we would expect them to receive multiple emails. I know it's not ideal but we've talked through it extensively with PS and they're ok with it. As for creating one send and using it for all relevant segments, that would be fine with me but I don't think it will work because the content of the email has to be dynamic based on the _segment_. For example, [the copy in the email templates mentions the CD](https://www.figma.com/design/1HrOm1tkJ3lowQhrSRkPLs/ZAP-Listserv%3A-Wireframes-and-User-Stories?node-id=190-28&node-type=frame&t=2Ojsd8uRqkYopgTn-0). 

Yeah, I figured there would be some that don't fall in any. That shouldn't be a concern.